src/main/java/gregtech/common/metatileentities/converter/MetaTileEntityConverter.java

```diff
@@ -0,0 +1,217 @@
+import codechicken.lib.render.CCRenderState;
+import codechicken.lib.render.pipeline.ColourMultiplier;
+import codechicken.lib.render.pipeline.IVertexOperation;
+import codechicken.lib.vec.Matrix4;
+import gregtech.api.GTValues;
+import gregtech.api.capability.FeCompat;
+import gregtech.api.capability.GregtechCapabilities;
+import gregtech.api.capability.tool.ISoftHammerItem;
+import gregtech.api.gui.ModularUI;
+import gregtech.api.metatileentity.MetaTileEntity;
+import gregtech.api.metatileentity.MetaTileEntityHolder;
+import gregtech.api.metatileentity.TieredMetaTileEntity;
+import gregtech.api.util.GTUtility;
+import gregtech.client.renderer.texture.Textures;
+import gregtech.client.utils.PipelineUtil;
+import gregtech.common.ConfigHolder;
+import gregtech.common.tools.DamageValues;
+import net.minecraft.client.renderer.texture.TextureAtlasSprite;
+import net.minecraft.client.resources.I18n;
+import net.minecraft.creativetab.CreativeTabs;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.ItemStack;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.EnumHand;
+import net.minecraft.util.NonNullList;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.text.TextComponentTranslation;
+import net.minecraft.world.World;
+import net.minecraftforge.common.capabilities.Capability;
+import net.minecraftforge.energy.CapabilityEnergy;
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.commons.lang3.tuple.Pair;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+import static gregtech.api.capability.GregtechDataCodes.SYNC_TILE_MODE;
+
+public class MetaTileEntityConverter extends TieredMetaTileEntity {
+
+    protected final ConverterTrait converterTrait;
+
+    protected final int amps;
+
+    public MetaTileEntityConverter(ResourceLocation metaTileEntityId, int tier, int amps) {
+        super(metaTileEntityId, tier);
+        this.amps = amps;
+        this.converterTrait = initializeTrait();
+        reinitializeEnergyContainer();
+    }
+
+    protected ConverterTrait initializeTrait() {
+        return new ConverterTrait(this, amps, true);
+    }
+
+    @Override
+    protected void reinitializeEnergyContainer() {
+        if (converterTrait == null) return;
+        this.energyContainer = converterTrait.getEnergyEUContainer();
+    }
+
+    @Override
+    public boolean onRightClick(EntityPlayer playerIn, EnumHand hand, EnumFacing facing, CuboidRayTraceResult hitResult) {
+        ItemStack itemStack = playerIn.getHeldItem(hand);
+        if (!itemStack.isEmpty() && itemStack.hasCapability(GregtechCapabilities.CAPABILITY_MALLET, null)) {
+            ISoftHammerItem softHammerItem = itemStack.getCapability(GregtechCapabilities.CAPABILITY_MALLET, null);
+
+            if (getWorld().isRemote) {
+                scheduleRenderUpdate();
+                return true;
+            }
```

Comment:
Should this check go after the check for if we can damage the soft hammer?

Reply:
This is the same order that things are done in for similar blocks like Diodes